src/pages/RepoPage/FailedTestsTab/FailedTestsTab.tsx

```diff
@@ -109,7 +106,63 @@ export default function FailedTestsTab() {
         </p>
       </div>
       <SetupOptionSelector />
-      <Content />
+      {usesCli ? <CodecovCLI /> : <GitHubActions />}
     </div>
   )
 }
+
+const Loader = () => (
+  <div className="flex justify-center py-8">
+    <Spinner />
+  </div>
+)
+
+interface URLParams {
+  provider: string
+  owner: string
+  repo: string
+}
+
+function FailedTestsTab() {
+  const { provider, owner, repo } = useParams<URLParams>()
+  const { data: repoOverview } = useRepoOverview({ provider, owner, repo })
+
+  if (repoOverview?.testAnalyticsEnabled) {
+    return (
+      <SentryRoute
+        path={[
+          '/:provider/:owner/:repo/tests',
+          '/:provider/:owner/:repo/tests/:branch',
+        ]}
+        exact
+      >
+        <Suspense fallback={<Loader />}>
+          <div className="flex flex-1 flex-col gap-4">
+            <FailedTestsTable />
+          </div>
+        </Suspense>
+      </SentryRoute>
+    )
+  }
+
+  return (
+    <Switch>
+      <SentryRoute path="/:provider/:owner/:repo/tests/new" exact>
+        <OnboardingWrapper usesCli={false} />
+      </SentryRoute>
+      <SentryRoute path="/:provider/:owner/:repo/tests/new/codecov-cli" exact>
+        <OnboardingWrapper usesCli />
+      </SentryRoute>
+      <Redirect
+        from={`/:provider/:owner/:repo/tests`}
+        to={`/:provider/:owner/:repo/tests/new`}
+      />
+      <Redirect
+        from={`/:provider/:owner/:repo/tests/*`}
```

Comment:
We should also add a redirect from `./tests/new/anythingelse` to `./tests/new`

Reply:
Updated in https://github.com/codecov/gazebo/pull/3042/commits/18984d4a734599b3a37b2fc89de0d9a269d73d90